CORE-8686 Add `toString`s to transaction implementations.

lankydan opened the pull request:
Also add some `equals` and `hashCode`s where appropriate.

Move a class to a different package.

lankydan committed:
CORE-8686 Add `toString`s to transaction implementations.

Also add some `equals` and `hashCode`s where appropriate.

Move a class to a different package.

vlajos reviewed (commented):
Objects.hash could be used in a few more places.
That could simplify the code slightly.

lankydan committed:
CORE-8686 Add `toString`s to transaction implementations.

Also add some `equals` and `hashCode`s where appropriate.

Move a class to a different package.

vlajos commented on components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/transaction/filtered/factory/UtxoFilteredTransactionFactoryImpl.kt:
```diff
@@ -23,6 +24,7 @@ import org.osgi.service.component.annotations.ServiceScope
 
 @Component(
     service = [UtxoFilteredTransactionFactory::class, UsedByFlow::class],
+    property = [SandboxConstants.CORDA_UNINJECTABLE_SERVICE],
```
Is this change intentionally in this PR?
(Otherwise all looks good.)

lankydan replied:
Yeah, to stop an error in the logs from popping up.

vlajos reviewed (approved):
LGTM
